src/main/java/htsjdk/samtools/cram/structure/block/Block.java

```diff
@@ -0,0 +1,234 @@
+     */
+    public static RawBlock buildNewFileHeaderBlock(final byte[] rawContent) {
+        return new RawBlock(BlockContentType.FILE_HEADER, rawContent);
+    }
+
+    /**
+     * Create a new compression header block with the given uncompressed content.
+     * The block will have RAW (no) compression and COMPRESSION_HEADER content type.
+     *
+     * @param rawContent the uncompressed content of the block
+     * @return a new compression header {@link RawBlock} object
+     */
+    public static RawBlock buildNewCompressionHeaderBlock(final byte[] rawContent) {
+        return new RawBlock(BlockContentType.COMPRESSION_HEADER, rawContent);
+    }
+
+    /**
+     * Create a new slice header block with the given uncompressed content.
+     * The block will have RAW (no) compression and MAPPED_SLICE content type.
+     *
+     * @param rawContent the uncompressed content of the block
+     * @return a new mapped slice {@link RawBlock} object
+     */
+    public static RawBlock buildNewSliceHeaderBlock(final byte[] rawContent) {
+        return new RawBlock(BlockContentType.MAPPED_SLICE, rawContent);
+    }
+
+    /**
+     * Create a new external block with the given uncompressed content, content ID and compression method.
+     * The block will have EXTERNAL content type.
+     *
+     * @param contentId the external identifier for the block
+     * @param compressor which external compressor to use on this block
+     * @param rawContent the uncompressed content of the block
+     * @return a new external {@link CompressibleBlock} object
+     */
+    public static CompressibleBlock buildNewExternalBlock(final int contentId, final ExternalCompressor compressor, final byte[] rawContent) {
+        if (contentId == Block.NO_CONTENT_ID) {
+            throw new CRAMException("Valid Content ID required.  Given: " + contentId);
+        }
+        
+        final byte[] compressedContent = compressor.compress(rawContent);
+        return new CompressibleBlock(compressor.getMethod(), BlockContentType.EXTERNAL, contentId, rawContent, compressedContent);
+    }
+
+    /**
+     * Create a new core block with the given uncompressed content.
+     * The block will have RAW (no) compression and CORE content type.
+     *
+     * @param rawContent the uncompressed content of the block
+     * @return a new mapped slice {@link RawBlock} object
+     */
+    public static RawBlock buildNewCoreBlock(final byte[] rawContent) {
+        return new RawBlock(BlockContentType.CORE, rawContent);
+    }
+
+    protected abstract BlockCompressionMethod getMethod();
+
+    /**
+     * Identifies whether this is a header or data block, and which kind
+     * @return the CRAM content type of the block
+     */
+    public final BlockContentType getContentType() {
+        return contentType;
+    }
+
+    /**
+     * Return the External Content ID for this block.
+     * By default this is NO_CONTENT_ID (0) but CompressibleBlocks override this
```

Comment:
CompressibleBlocks can still have Content ID 0 if they are not external.  Should I clarify that here?